api/STRUCTURE.md

```diff
@@ -0,0 +1,83 @@
+
+TODO: do we have a preference of which of those to do?
+
+## DB Models
+
+DB Models represent a single row of a single table in the database, possibly with extender tables attached. We denote DB models by prefixing `Db` to the name of the database table. DB models heavily use `javax.persistence` annotations to describe where in the database table each part of the model lives.
+
+In order to automatically gather data from extender tables, we use the `@JoinColumn`, `@OneToOne`, `@OneToMany`, `@ManyToOne`, and `@MappedBy` [annotations](https://www.baeldung.com/jpa-join-column). When using these annotations, be sure to carefully consider which table should 'own' the relationship. With a `@ManyToOne` relationship, the `@ManyToOne` side should generally own the relationship. For example. if we have an extender table containing a user's addresses, `DbAddress` should have a `@ManyToOne` `DbUser` member and `DbUser` should have a `@OneToMany` `DbAddress` member, with the annotation containing a `mappedBy` attribute pointing to `DbAddress`'s `DbUser` member as the member defining the relationship.
+
+TODO: wording on the above is Bad. pls help
+
+Much of the structure of DB models can be autogenerated in IntelliJ by defining the class members and then doing Generate -> Getter and Setter and Generate -> `equals()` and `hashCode()`. This will not automatically add the required annotations - those must be added manually.
+
+## Services and ServiceImpls
+
+Services are interfaces that declare methods of business logic. ServiceImpls are classes implementing their correspondingly named Service and the business logic methods declared therein.
+
+TODO: Why do we do this this way?
+
+Services / ServiceImpls are generally named after a feature or functional area, like 'UserService', 'CohortService', or 'MonitoringService'.
+
+ServiceImpls can call into other services, but we try to keep the number of dependencies down. We try to avoid having ServiceImpls call into a large number of DAOs, especially DAOs not directly related to the functional area for which the ServiceImpl is responsible. For instance, the `UserServiceImpl` calls into the `UserDao`, the `AdminActionHistoryDao`
+
+ServiceImpls should handle only application logic. They should not need explicit knowledge of the database, the REST API, or the UI.
+
+## Controllers
+
+Controllers take Request objects from the API, call into the Service layer, and return Response objects back to the API. Controllers implement autogenerated [APIControllers](#apicontrollers).
+
+Controller methods should generally be small and should not contain any actual logic. The most common conditional in the Controller layer should be checking a feature flag. Most of the work done by Controllers should be munging data provided by the API or a Service.
```

Comment:
> and should not contain any actual logic
💯 

> munging data provided by the API or a Service
I'd say "packaging results into a ResponseEntity" and possibly something about translating errors (occasionally).